File: app/src/main/java/com/project/poupay/sql/Preferences.java
package com.project.poupay.sql;

import static android.content.Context.MODE_PRIVATE;

import android.content.Context;

public class Preferences {
    private static final String prefsId = "com.project.poupay";

    public static final String REMIND_LOGIN_ENABLED = "REMIND_LOGIN_ENABLED";
    public static final String REMIND_LOGIN_USERNAME = "REMIND_LOGIN_USERNAME";
    public static final String REMIND_LOGIN_PASSWORD = "REMIND_LOGIN_PASSWORD";

    public static void set(String id, boolean value, Context context) {
        context.getSharedPreferences(prefsId, MODE_PRIVATE)
                .edit()
                .putBoolean(id, value)
                .apply();
    }

    public static void set(String id, String value, Context context) {
        context.getSharedPreferences(prefsId, MODE_PRIVATE)
                .edit()
                .putString(id, value)
                .apply();
    }

    public static void set(String id, int value, Context context) {
        context.getSharedPreferences(prefsId, MODE_PRIVATE)
                .edit()
                .putInt(id, value)
                .apply();
    }

    public static int getInt(String id, int defValue, Context context) {
        return context.getSharedPreferences(prefsId, MODE_PRIVATE)
                .getInt(id, defValue);
    }


    public static boolean getBool(String id, boolean defValue, Context context) {
        return context.getSharedPreferences(prefsId, MODE_PRIVATE)
                .getBoolean(id, defValue);
    }

    public static String getString(String id, String defValue, Context context) {
        return context.getSharedPreferences(prefsId, MODE_PRIVATE)
                .getString(id, defValue);
    }

}

File: app/src/main/java/com/projeto/poupay/sql/Preferences.kt
package com.projeto.poupay.sql

import android.content.Context
import android.content.Context.MODE_PRIVATE

class Preferences {
    enum class Entry {
        REMIND_LOGIN_ENABLED,
        REMIND_LOGIN_USERNAME,
        REMIND_LOGIN_PASSWORD
    }

    companion object {
        private var prefsId = "com.project.poupay.${SqlQueries.username}."

        fun set(entry: Entry, value: Boolean, context: Context) {
            context.getSharedPreferences(prefsId, MODE_PRIVATE)
                .edit()
                .putBoolean(entry.toString(), value)
                .apply()
        }

        fun set(entry: Entry, value: String, context: Context) {
            context.getSharedPreferences(prefsId, MODE_PRIVATE)
                .edit()
                .putString(entry.toString(), value)
                .apply()
        }

        fun set(entry: Entry, value: Int, context: Context) {
            context.getSharedPreferences(prefsId, MODE_PRIVATE)
                .edit()
                .putInt(entry.toString(), value)
                .apply()
        }

        fun getBoolean(entry: Entry, defValue: Boolean, context: Context) = context.getSharedPreferences(prefsId, MODE_PRIVATE).getBoolean(entry.toString(), defValue)

        fun getString(entry: Entry, defValue: String, context: Context) = context.getSharedPreferences(prefsId, MODE_PRIVATE).getString(entry.toString(), defValue)

    }


}